### ISSUE TITLE
[FEATURE] A11y improvements

### ISSUE THREAD

**Issue by dingo-d** [infinum/eightshift-frontend-libs#170]
There are a couple of accessibility improvements we should implement in the libs, especially blocks.

* Focus and hover states - simple color change is not accessible, especially if the color contrast change isn't set properly
  "Oh, but outlines are ugly" I hear you say - well imagine being visually impaired and having to discern if something is focused or hovered or not. Not a nice experience. The client design can remove those, but we should have them by default (although we should educate the client on the importance of those, even though they are 'ugly'). The same is with the underlined links.
* Missing aria labels - when an anchor that is used for logos, or toggle buttons doesn't contain any content we should add `aria-label` so that screen readers can recognize what the link is used for

**OK**:

A case where `aria-label` is not necessary, because there is content inside anchor tag, and screen readers can read it

```html
<a href="https://www.google.com">Go to google</a>
```

No content case, but with `aria-label` attribute

```html
<a href="#" aria-label="Mobile menu toggle" class="menu-toggle js-menu-toggle"></a>
```

**Not OK**:

```html
<a href="#" class="menu-toggle js-menu-toggle"></a>
```

* Alt tags on image elements - all the components that have any kind of image renders should pull the `alt` tag from the image set in the media library
* When images are used as a background image CSS, `aria-label` should be provided instead of `alt` with added `role="img"` attribute, but **only** when the background image is used to display a meaningful image.  
If it's used only for **decorative purposes** (parallax or hero image) those attributes **shouldn't** be added.  
If you have an image in the text that is set up using CSS (for whatever reason), that container should have the `role` and `aria-label` to it.
Also, if the bg image has content inside but is not decorative `role` and `aria-label` should be placed in a separate `span` item

**OK**:

```html
<div>
<span class="background-image" role="img" aria-label="[place alt text here]> </span>
[all the rest of my content]
</div>
```

**Not OK**:

```html
<div class="background-image" role="img" aria-label="blah blah blah">
[all the rest of my content]
</div>
```

* Contrast improvements - check if the default colors are satisfying at least contrast ratio of 4.5:1 ([link](https://www.a11yproject.com/posts/2015-01-05-what-is-color-contrast/#:~:text=A%20contrast%20ratio%20of%204.5,Reading%20section%20following%20the%20article.))
* Make sure all the content can be navigated through using the keyboard
* Add [skip-link](https://make.wordpress.org/themes/handbook/review/required/#skip-links) - this one is probably something we should add to boilerplate theme since it's theme-specific. Plugins don't need them.

These are all that came up to me off the top of my head. They don't seem like much but make people using our tools increasing web accessibility a bit. And we should all strive to make the web more accessible 🙂

**Comment by mbmjertan** [comment]
I also noticed that on some projects (all of them?) we don't respect the [Reduce motion feature](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion). I'll add other noticed a11y issues here as well.

**Comment by iruzevic** [comment]
dulicate
